pkg/shp/cmd/buildrun/logs.go

```diff
@@ -61,29 +77,50 @@ func (c *LogsCommand) Run(params *params.Params, ioStreams *genericclioptions.IO
 		LabelSelector: fmt.Sprintf("%v=%v", buildv1alpha1.LabelBuildRun, c.name),
 	}
 
+	// first see if pod is already done; if so, even if we have follow == true, just do the normal path;
+	// we don't employ a pod watch here since the buildrun may already be complete before 'shp buildrun logs -F'
+	// is invoked.
+	justGetLogs := false
 	var pods *corev1.PodList
-	if pods, err = clientset.CoreV1().Pods(params.Namespace()).List(c.cmd.Context(), lo); err != nil {
+	err = wait.PollImmediate(1*time.Second, 10*time.Second, func() (done bool, err error) {
+		if pods, err = clientset.CoreV1().Pods(params.Namespace()).List(c.cmd.Context(), lo); err != nil {
+			fmt.Fprintf(ioStreams.ErrOut, "error listing Pods for BuildRun %q: %s\n", c.name, err.Error())
+			return false, nil
+		}
+		if len(pods.Items) == 0 {
+			fmt.Fprintf(ioStreams.ErrOut, "no builder pod found for BuildRun %q\n", c.name)
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
 		return err
 	}
-	if len(pods.Items) == 0 {
-		return fmt.Errorf("no builder pod found for BuildRun %q", c.name)
+	pod := pods.Items[0]
+	phase := pod.Status.Phase
+	if phase == corev1.PodFailed || phase == corev1.PodSucceeded {
+		justGetLogs = true
 	}
 
-	fmt.Fprintf(ioStreams.Out, "Obtaining logs for BuildRun %q\n\n", c.name)
+	if !c.follow || justGetLogs {
+		fmt.Fprintf(ioStreams.Out, "Obtaining logs for BuildRun %q\n\n", c.name)
 
-	var b strings.Builder
-	pod := pods.Items[0]
-	for _, container := range pod.Spec.Containers {
-		logs, err := util.GetPodLogs(c.cmd.Context(), clientset, pod, container.Name)
-		if err != nil {
-			return err
+		var b strings.Builder
+		for _, container := range pod.Spec.Containers {
```

Comment:
In follow.go, you include initContainers, should we include them here as well for consistency?

Reply:
yep good catch I'll add